persistence.js

```diff
@@ -29,6 +29,7 @@ function matchTopic (p) {
 }
 
 MemoryPersistence.prototype.storeRetained = function (packet, cb) {
+  packet = Object.assign({}, packet)
```

Comment:
Can you add a check in a unit test for this?
Also, you should use `new Packet(packet)`, it should work the same.